syntax="proto3";

package user;

service UserService {
    rpc CreateUser (CreateUserRequest) returns (UserResponse);
    rpc ReadUser (ReadUserRequest) returns (UserResponse);
    rpc UpdateUser (UpdateUserRequest) returns (UserResponse);
    rpc DeleteUser (DeleteUserRequest) returns (UserResponse);
    rpc ReadAll (ReadAllRequest) returns (ReadAllResponse);
}

message CreateUserRequest {
    int32 id = 1;
    string name = 2;
    string address = 3;
    int32 salary = 4;
    string login_id = 5;
    string password = 6;
}

message ReadUserRequest {
    int32 id = 1;
}

message UpdateUserRequest {
    int32 id = 1;
    string name = 2;
    string address = 3;
    int32 salary = 4;
    string login_id = 5;
    string password = 6;
}

message DeleteUserRequest {
    int32 id = 1;
}

message UserResponse {
    string result = 1;
    string error = 2;
}

message ReadAllRequest{}

message User {
    int32 id = 1;
    string name = 2;
    string address = 3;
    int32 salary = 4;
    string login_id = 5;
    string password = 6;
}

message ReadAllResponse{
    repeated User users = 1;
}
